Src/IronPython/Runtime/ByteArray.cs

```diff
@@ -301,6 +301,14 @@ public ByteArray center(int width, [BytesLike, NotNull]IList<byte> fillchar)
             }
         }
 
+        public bool endswith([BytesLike, NotNull]IList<byte> suffix, object? start = null, object? end = null) {
```

Comment:
I'll fix that and add some tests to catch it. There are more methods that need an upgrade like this so it is good to have a complete solution in the first place and then replicate it.

Reply:
I can't figure it out how to make it rejecting the keyword arguments. What is the right way of doing it? I thought that by removing the default parameter values and providing explicit overloads for all allowed variants would do the trick, but it only disallows keywords if the preceding argument is skipped.

Here is an example using `bytes.count`, which does not define any default parameter values, and in CPython is defined as `bytes.count(sub[, start[, end]]) -> int`:

```
>>> b'abc'.count(b'a', 0)
1
>>> b'abc'.count(b'a', 0, 3)
1
>>> b'abc'.count(b'a', 0, end=3)
1
>>> b'abc'.count(b'a', start=0, end=3)
1
>>> b'abc'.count(b'a', end=3, start=0)
1
>>> b'abc'.count(b'a', start=0)
1
>>> b'abc'.count(b'a', end=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: count() got an unexpected keyword argument 'end'
```

Reply:
Interesting, I also thought having separate overloads would do the trick but it looks like it will accept keyword arguments for the overload with the appropriate number of arguments? I guess this is an OK behavior...

I will amend my comment to be that we should probably not specify a default value if CPython does not define one (even though it appears to be equivalent to passing in null in this case).

Reply:
It is basically a C# behavior, perhaps unsurprisingly. But it does not match CPython:

```
>>> b'abc'.count(b'a', start=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: count() takes no keyword arguments
```

However, as I am reading more about [issues with argument handling in CPython](https://www.python.org/dev/peps/pep-0436/), it appears that this is more a legacy than some desirable effect. So I guess it is OK...

OK to the rule not to specify defaults if CPython doesn't define them. The effect will be then limited to the docstring generated by `help` and rejecting a case when an argument is provided without providing all preceding arguments.